### ISSUE TITLE
Add way to get the compiled nested discriminator models

### ISSUE THREAD

**Issue by hasezoey** [typegoose/typegoose#744]
## Describe what you need | want

Currently nested discriminator return values are discarded and would be useful for `instanceof` checks (or other things), but typegoose currently does not provide a way to get them

## Do you have already an idea for the implementation?

maybe store the returned models in a `Map<string, Map<string, mongoose.Model>>` (nested map) where the first map key is `${finalModelName}.${key}` and the key for the second map is `${finalChildName}`
OR
patch in a symbol (which has the nested map of the example above) into the model and a function to get from what symbol instead of a global map

**Comment by hasezoey** [comment]
a quick patch would be something like:
```ts
const ret = path.discriminator(getName(child), childSch, finalChildName);
const asName = `${finalName}.${key}`;
let got = testData.get(asName) as Map<string, mongoose.Model<any>>;

if (isNullOrUndefined(got)) {
  got = new Map();
  testData.set(asName, got);
}

got.set(finalChildName, ret);
```

**Comment by hasezoey** [comment]
Mongoose already natively provides a list for nested discriminators, see https://github.com/Automattic/mongoose/discussions/12159

TL;DR: `schema.path('nestedpath').caster(Constructor).discriminators` (`caster` for `SubDocuments`, `casterConstructor` for `SubDocumentArrays`)

and i dont think typegoose can provide better coverage than this - at least i dont know a better way than this (feel free the propose a better thing)
